compiler/pash_runtime_daemon.py

```diff
@@ -262,10 +270,53 @@ def run(self, in_filename, out_filename):
                 self.fout = open(out_filename, "w")
 
             self.parse_and_run_cmd(input_cmd)
-            
+        
+        self.reader.close()
         shutdown()
 
 
+class UnixPipeReader:
+    def __init__(self, in_filename, blocking = True):
+        self.in_filename = in_filename
+        self.buffer = ""
+        self.blocking = blocking
+        if not self.blocking:
```

Comment:
Log whether the initialization was blocking or non-blocking (not that we will use the non-blocking API).

Reply:
Actually, we should add a comment that non-blocking is not supposed to be used. And that it was just an experiment.